jupyter_server_nbmodel/handlers.py

```diff
@@ -219,15 +228,32 @@ async def _execute_snippet(
         The execution status and outputs.
     """
     ycell = None
+    time_info = {}
     if metadata is not None:
         ycell = await _get_ycell(ydoc, metadata)
         if ycell is not None:
+            execution_start_time = datetime.now(timezone.utc).isoformat()[:-6]
             # Reset cell
             with ycell.doc.transaction():
                 del ycell["outputs"][:]
                 ycell["execution_count"] = None
                 ycell["execution_state"] = "running"
-
+                if "execution" in ycell["metadata"]:
+                    del ycell["metadata"]["execution"]
+                if metadata.get("record_timing", False):
+                    time_info = ycell["metadata"].get("execution", {})
+                    time_info["shell.execute_reply.started"] = execution_start_time
```

Comment:
It looks that we also need to set `iopub.execute_input` for compatibility with `jupyterlab-execute-time` so that it can parse notebooks after refresh. There is a subtle difference between these two but I think for now allowing users to work with that extension would be a big win:

```suggestion
                    time_info["shell.execute_reply.started"] = execution_start_time
                    time_info["iopub.execute_input"] = execution_start_time
```